src/inmanta/ast/type.py

```diff
@@ -276,11 +301,17 @@ def __init__(self) -> None:
         self.try_cast_functions: Sequence[Callable[[Optional[object]], object]] = [int]
 
     def validate(self, value: Optional[object]) -> bool:
-        if not super().validate(value):
-            return False
-        if not isinstance(value, numbers.Integral):
+        """
+        Validate the given value to check if it satisfies the constraints
+        associated with this type
+        """
+        if isinstance(value, AnyType):
+            return True
+
+        if not isinstance(value, int):
```

Comment:
What does the change from `numbers.Ingegral` -> `int` entail? Is this breaking?

Reply:
I must say I have no idea, I didn't think of it when changing this. I'll revert to numbers.Integral, it makes me more comfortable